src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1509,6 +1509,38 @@ public void MainWindowHandle_GetNotStarted_ThrowsInvalidOperationException()
             var process = new Process();
             Assert.Throws<InvalidOperationException>(() => process.MainWindowHandle);
         }
+        
+        [Fact(Skip = "Manual test")]
```

Comment:
We don't add tests like this; it'll never get run and end up just being dead code.

What prevents writing an automated test that's not permanently disabled?

Reply:
The only way I can think of to test the full extend of this (i.e. whether `MainWindowHandle` eventually becomes non-zero) is to spawn a process with a UI. Based on [this comment](https://github.com/dotnet/runtime/issues/32690#issuecomment-589918826) from @danmosemsft, I was assuming that this is undesirable for the pipeline, but it was fine to have a test like that if marked as manual.

However, there is an alternative: rather than checking if `MainWindowHandle` became non-zero, I could set it to a non-zero value, then check if it was refreshed back to zero. Using reflection, the test could look like this:

```csharp
[Fact]
[PlatformSpecific(TestPlatforms.Windows)]
public void MainWindowHandle_ShouldRefresh_Windows()
{
    FieldInfo handleField = typeof(Process).GetField("_mainWindowHandle", BindingFlags.Instance | BindingFlags.NonPublic);
    FieldInfo flagField = typeof(Process).GetField("_haveMainWindow", BindingFlags.Instance | BindingFlags.NonPublic);

    CreateDefaultProcess();

    Assert.Equal(IntPtr.Zero, _process.MainWindowHandle);

    // MainWindowHandle should not update if _haveMainWindow is true until the process is refreshed
    IntPtr fakeHandle = (IntPtr)1234;
    handleField.SetValue(_process, fakeHandle);
    flagField.SetValue(_process, true);
    Assert.Equal(fakeHandle, _process.MainWindowHandle);
    _process.Refresh();
    Assert.Equal(IntPtr.Zero, _process.MainWindowHandle);
}
```

Instead of using reflection, I could also make use of `InternalsVisibleToAttribute` (as I see that other libraries do that already), and define the `_haveMainWindow` and `_mainWindowHandle` fields as internal. In both cases, the test might be getting too deep into the implementation details. As a slight variant, I could also add an internal setter to `MainWindowHandle` and an internal property `HaveMainWindow`, or perhaps an internal method to set those two fields.

Let me know which approach would you prefer and I will update the PR accordingly.

Reply:
I'd be fine if the test is attributed as both `[Outerloop]` and `[Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)]`.  Then it'll only affect a dev working locally and running outerloop tests, in which case I think popping UI here and there is fair game... we already do it in the Process tests IIRC.

Reply:
+1

Reply:
No problem, I did the changes as requested. The job "Libraries Test Run release mono OSX x64 Debug" failed, but I would think it is unrelated to the changes, as they should only affect Windows in any case. Is it [this same issue](https://github.com/dotnet/runtime/issues/32256) perhaps?